src/nectarchain/calibration/NectarGain/SPEfit/utils.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import math
+
+import numpy as np
+from iminuit import Minuit
+from scipy import interpolate, signal
+from scipy.special import gammainc
+
+logging.basicConfig(format='%(asctime)s %(name)s %(levelname)s %(message)s')
+log = logging.getLogger(__name__)
+log.handlers = logging.getLogger('__main__').handlers
+
+__all__ = ['UtilsMinuit','Multiprocessing']
+from .parameters import Parameters
+
+class Multiprocessing() : 
+    @staticmethod
+    def custom_error_callback(error):
+        log.error(f'Got an error: {error}')
+        log.error(error,exc_info=True)
+
+class UtilsMinuit() :
+    @staticmethod
+    def make_minuit_par_kwargs(parameters : Parameters):
+        """Create *Parameter Keyword Arguments* for the `Minuit` constructor.
+
+        updated for Minuit >2.0
+        """
+        names = parameters.parnames
+        kwargs = {"names": names,"values" : {}}
+
+        for parameter in parameters.parameters:
+            kwargs["values"][parameter.name] = parameter.value
+            min_ = None if np.isnan(parameter.min) else parameter.min
+            max_ = None if np.isnan(parameter.max) else parameter.max
+            error = 0.1 if np.isnan(parameter.error) else parameter.error
+            kwargs[f"limit_{parameter.name}"] = (min_, max_)
+            kwargs[f"error_{parameter.name}"] = error
+        return kwargs
+    
+    @staticmethod
+    def set_minuit_parameters_limits_and_errors(m : Minuit,parameters : dict) :
+        """function to set minuit parameter limits and errors with Minuit >2.0
+
+        Args:
+            m (Minuit): a Minuit instance
+            parameters (dict): dict containing parameters names, limits errors and values
+        """
+        for name in parameters["names"] :
+            m.limits[name] = parameters[f"limit_{name}"]
+            m.errors[name] = parameters[f"error_{name}"]
+
+
+# Usefull fucntions for the fit
+def gaussian(x, mu, sig):
+    return (1./(sig*np.sqrt(2*math.pi)))*np.exp(-np.power(x - mu, 2.) / (2 * np.power(sig, 2.)))
+
+def weight_gaussian(x,N, mu, sig) : 
+    return N * gaussian(x, mu, sig)
+
+def doubleGauss(x,sig1,mu2,sig2,p):
+    return p *2 *gaussian(x, 0, sig1) + (1-p) * gaussian(x, mu2, sig2)
+
+def PMax(r):
+    if (np.pi*r**2/(np.pi*r**2 + np.pi - 2*r**2 - 2) <= 1):
+        return np.pi*r**2/(np.pi*r**2 + np.pi - 2*r**2 - 2)
```

Comment:
Perhaps you can store the result of this computation in a variable for readability purpose (and to avoid recompute things twice) :
pmax = np.pi*r**2/(np.pi*r**2 + np.pi - 2*r**2 - 2)

Reply:
Indeed it is better, I change that